src/internal/k8s/tunnel.go

```diff
@@ -141,6 +136,9 @@ func (tunnel *Tunnel) Connect(target string, blocking bool) {
 			}
 		}
 
+		// Dump the tunnel URL to the console for other tools to use
+		fmt.Print(url)
```

Comment:
Given that the spinner logs this url and that the function returns the url to the calling function, I don't think we need to be printing the url to stdout too.

Reply:
This is a particular issue for needing to push to `stdout` vs `stderr`, I agree I don't _love this_, but functionally speaking, we really needed to do this if someone wanted to script out `zarf connect registry` for example now that the ports aren't predictable.  I suppose you could argue they should just use a specific port like we do for tests, but then they'd need to guarantee port availability first and won't have the logic we do already handy.

Reply:
Very much open to ideas on that.

Reply:
What if we hold the print until we return back to the `connect.go` command and only print if `--cli-only` is provided (or make a new flag that is explicit for returning the port to stdout?

Reply:
Updated to meet the intent of what you asked, could not push it back to the `cmd` package because the call needs to be blocking or the tunnel would die.

Reply:
That works for me.

Sometime in the future I might propose moving the logic to handle when the `local-port == 0` to the `k8s.NewTunnel()` function instead of happening inside the `tunnel.Connect()` function that way we can print it out in the `cmd.Connect()` and just be more explicit in general.

Reply:
Yeah that's a totally valid point too, I think there's still more cleanup overall in that `cmd.Connect()` area.